### ISSUE TITLE
W3664 false positive: regional CloudWatch Logs principal (`logs.<region>.amazonaws.com`) rejected since v1.49.0

### ISSUE THREAD

**Issue by jasperges** [aws-cloudformation/cfn-lint#4504]
### CloudFormation Lint Version

v1.50.1

### What operating system are you using?

Fedora

### Describe the bug

W3664 flags `logs.eu-west-1.amazonaws.com` as an invalid principal when `SourceArn` references an `AWS::Logs::LogGroup`, insisting on `logs.amazonaws.com` instead.

The regional principal form has been valid for CloudWatch Logs subscription filter permissions for years and still appears in current AWS documentation (e.g. the [AWS SDK for Java PutSubscriptionFilter example](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/example_cloudwatch-logs_PutSubscriptionFilter_section.html)).

The W3664 rule was introduced in v1.48.0 and still accepted both forms. The regression was introduced in v1.49.0 by commit `2cd1f2919b48342d0d6899fb85f3dcae6832d8c9` ([PR #4444](https://github.com/aws-cloudformation/cfn-lint/pull/4444)), which added a strict exact-match entry to `src/cfnlint/data/schemas/extensions/aws_lambda_permission/permission_principal.json`:
```json
"AWS::Logs::LogGroup": "logs.amazonaws.com",
```

https://github.com/aws-cloudformation/cfn-lint/pull/4444/changes#diff-09cbd3859e91e4c31751c4b8affd43b60f501410c6a2c9ec0b8d05e346da2f4fR40

### Expected behavior

Both `logs.amazonaws.com` and `logs.<region>.amazonaws.com` should be accepted as valid principals.

### Reproduction template

```yaml
AWSTemplateFormatVersion: 2010-09-09
Description: Reproduces W3664 false positive for regional CWL principal (regression since v1.49.0)

Resources:
  MyFunction:
    Type: AWS::Lambda::Function
    Properties:
      FunctionName: my-log-processor
      Handler: index.handler
      Runtime: python3.12
      Role: !Sub "arn:aws:iam::${AWS::AccountId}:role/my-execution-role"
      Code:
        ZipFile: |
          def handler(event, context):
              print(event)

  MyLogGroup:
    Type: AWS::Logs::LogGroup
    Properties:
      LogGroupName: /aws/my-app/logs
      RetentionInDays: 30

  MySubscriptionPermission:
    Type: AWS::Lambda::Permission
    Properties:
      FunctionName: !GetAtt MyFunction.Arn
      Action: lambda:InvokeFunction
      Principal: logs.eu-west-1.amazonaws.com  # regional principal — valid, but flagged since v1.49.0
      SourceArn: !GetAtt MyLogGroup.Arn

  MySubscriptionFilter:
    Type: AWS::Logs::SubscriptionFilter
    DependsOn: MySubscriptionPermission
    Properties:
      LogGroupName: !Ref MyLogGroup
      FilterPattern: ""
      DestinationArn: !GetAtt MyFunction.Arn
```
